mobile/lib/modules/home/services/asset_cache.service.dart

```diff
@@ -0,0 +1,85 @@
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:collection/collection.dart';
+import 'package:flutter/foundation.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:http/http.dart';
+import 'package:openapi/api.dart';
+import 'package:path_provider/path_provider.dart';
+
+abstract class JsonCache<T> {
+  final String cacheFileName;
+
+  JsonCache(this.cacheFileName);
+
+  Future<File> _getCacheFile() async {
+    final basePath = await getTemporaryDirectory();
+    final basePathName = basePath.path;
+
+    final file = File("$basePathName/$cacheFileName.bin");
```

Comment:
What is the advantage of this route vs going with the Hive integration for caching?

Reply:
I was trying both approaches (see commit history). The problem is that large hive boxes take long time to open (in the order of multiple seconds). That delays the app start.

Another advantage is that `getTemporaryDirectory()` is considered cache storage so the OS may clear this data if it's low on storage.
Also it shows as cache storage on Android and can be cleared by the user.

Reply:
Got it